### PR TITLE
add tests for a pinned flow

### PR DESCRIPTION
**Pull request by andriidovhan**
ADD tests: pinned flow

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+        def currentPath = pathHelper.convert(northbound.getFlowPath(flow.id))
+
+        when: "Make another path more preferable"
+        def newPath = switchPair.paths.find { it != currentPath }
+        switchPair.paths.findAll { it != newPath }.each { pathHelper.makePathMorePreferable(newPath, it) }
+
+        and: "Init reroute(manually)"
+        def isl = pathHelper.getInvolvedIsls(currentPath).first()
+        northbound.rerouteLinkFlows(isl.srcSwitch.dpId, isl.srcPort, isl.dstSwitch.dpId, isl.dstPort)
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowSpec.groovy]: why not `northbound.rerouteFlow(flow.id)` ? This is fine, but just wondering

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            }*.meterId]
+        }
+
+        cookiesMap.sort() == cookiesMapAfterReroute.sort()
+        metersMap.sort() == metersMapAfterReroute.sort()
+
+        when: "The broken ISL is restored"
+        northbound.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+
+        then: "The flow is marked as UP and not rerouted"
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowSpec.groovy]: I think the alt path should be made more preferable than the current path in order to properly verify this

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowSpec.groovy]: no specific reason.
just tried to do it in unusual way

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowSpec.groovy]: added

**Pull request closed without merging**